CI: Travis: panic: close of closed channel in (*IPMasqAgent).Start

Seen in PR #11368 posted 3 hours ago:
https://travis-ci.com/github/cilium/cilium/jobs/328758821

```
=== RUN   Test
panic: close of closed channel
goroutine 27 [running]:
github.com/cilium/cilium/pkg/ipmasq.(*IPMasqAgent).Start.func1(0xc000032af0)
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipmasq/ipmasq.go:146 +0x368
created by github.com/cilium/cilium/pkg/ipmasq.(*IPMasqAgent).Start
	/home/travis/gopath/src/github.com/cilium/cilium/pkg/ipmasq/ipmasq.go:124 +0x3cb
FAIL	github.com/cilium/cilium/pkg/ipmasq	0.350s
FAIL
Makefile:151: recipe for target 'unit-tests' failed
make: *** [unit-tests] Error 1
The command "./.travis/build.sh" exited with 2.
```

I'm not sure whether this is with or without PR https://github.com/cilium/cilium/pull/11317

--- a/pkg/ipmasq/ipmasq_test.go
+++ b/pkg/ipmasq/ipmasq_test.go
@@ -165,9 +165,8 @@ func (i *IPMasqTestSuite) TestUpdate(c *check.C) {
 }
 
 func (i *IPMasqTestSuite) TestRestore(c *check.C) {
-	// Stop ip-masq-agent goroutine (we can't use i.ipMasqMap.Stop(), as it stops
-	// the watcher)
-	close(i.ipMasqAgent.stop)
+	// Check that stale entry is removed from the map after restore
+	i.ipMasqAgent.Stop()
 
 	_, cidr, _ := net.ParseCIDR("3.3.3.0/24")
 	i.ipMasqMap.cidrs[cidr.String()] = *cidr
@@ -177,6 +176,8 @@ func (i *IPMasqTestSuite) TestRestore(c *check.C) {
 	_, err := i.configFile.WriteString("nonMasqueradeCIDRs:\n- 4.4.0.0/16")
 	c.Assert(err, check.IsNil)
 
+	i.ipMasqAgent, err = newIPMasqAgent(i.configFile.Name(), i.ipMasqMap)
+	c.Assert(err, check.IsNil)
 	i.ipMasqAgent.Start()
 	time.Sleep(300 * time.Millisecond)
 
@@ -187,12 +188,14 @@ func (i *IPMasqTestSuite) TestRestore(c *check.C) {
 
 	// Now stop the goroutine, and also remove the maps. It should bootstrap from
 	// the config
-	close(i.ipMasqAgent.stop)
+	i.ipMasqAgent.Stop()
 	i.ipMasqMap = &ipMasqMapMock{cidrs: map[string]net.IPNet{}}
 	i.ipMasqAgent.ipMasqMap = i.ipMasqMap
 	_, err = i.configFile.Seek(0, 0)
 	c.Assert(err, check.IsNil)
 	_, err = i.configFile.WriteString("nonMasqueradeCIDRs:\n- 3.3.0.0/16")
+	c.Assert(err, check.IsNil)
+	i.ipMasqAgent, err = newIPMasqAgent(i.configFile.Name(), i.ipMasqMap)
 	c.Assert(err, check.IsNil)
 	i.ipMasqAgent.Start()
 

